File: src/main/java/minic/ui/MiniCHoverInspector.java
package minic.ui;

import javafx.beans.property.ObjectProperty;
import javafx.beans.property.ReadOnlyObjectProperty;
import javafx.beans.property.SimpleObjectProperty;

import java.util.Objects;

/**
 * Graph View hover inspector 的共享状态。
 */
public final class MiniCHoverInspector {
    private final ObjectProperty<MiniCHoverInspectorContent> content =
            new SimpleObjectProperty<>(MiniCHoverInspectorContent.empty());

    /**
     * 当前内容属性。
     *
     * @return 只读内容属性
     */
    public ReadOnlyObjectProperty<MiniCHoverInspectorContent> contentProperty() {
        return content;
    }

    /**
     * 显示 hover 内容。
     *
     * @param content 内容
     */
    public void show(MiniCHoverInspectorContent content) {
        this.content.set(Objects.requireNonNull(content, "content"));
    }

    /**
     * 清空 hover 内容。
     */
    public void clear() {
        content.set(MiniCHoverInspectorContent.empty());
    }
}

File: uiweb/src/panel/MiniCHoverInspector.ts
import type { JavaMirrorFile } from "../translation/javaMirror";
import { MiniCHoverInspectorContent } from "./MiniCHoverInspectorContent";

export const miniCHoverInspectorMirror = {
  "javaPath": "src/main/java/minic/uilocal/panel/MiniCHoverInspector.java",
  "webPath": "uiweb/src/panel/MiniCHoverInspector.ts",
  "packageName": "minic.uilocal",
  "exportName": "MiniCHoverInspector",
  "kind": "class",
  "imports": [
    "javafx.beans.property.ObjectProperty",
    "javafx.beans.property.ReadOnlyObjectProperty",
    "javafx.beans.property.SimpleObjectProperty",
    "java.util.Objects"
  ],
  "fields": [
    {
      "name": "content",
      "signature": "private final ObjectProperty<MiniCHoverInspectorContent> content ="
    }
  ],
  "methods": [
    {
      "name": "contentProperty",
      "signature": "contentProperty()"
    },
    {
      "name": "show",
      "signature": "show(MiniCHoverInspectorContent content)"
    },
    {
      "name": "clear",
      "signature": "clear()"
    }
  ]
} as const satisfies JavaMirrorFile;

export class MiniCHoverInspector {
  static readonly mirror = miniCHoverInspectorMirror;

  readonly mirror = miniCHoverInspectorMirror;

  private content = MiniCHoverInspectorContent.empty();

  private readonly listeners = new Set<(content: MiniCHoverInspectorContent) => void>();

  contentProperty(): MiniCHoverInspectorContent {
    return this.content;
  }

  show(content: MiniCHoverInspectorContent): void {
    this.setContent(content);
  }

  clear(): void {
    this.setContent(MiniCHoverInspectorContent.empty());
  }

  subscribe(listener: (content: MiniCHoverInspectorContent) => void): () => void {
    this.listeners.add(listener);
    return () => {
      this.listeners.delete(listener);
    };
  }

  summary(): string {
    return `MiniCHoverInspector: ${this.mirror.methods.length} methods, ${this.mirror.fields.length} fields`;
  }

  private setContent(content: MiniCHoverInspectorContent): void {
    this.content = content;
    this.listeners.forEach((listener) => listener(content));
  }
}

export default MiniCHoverInspector;
